File: Check_If_It_Is_A_Straight_Line.py
class Solution:
    def checkStraightLine(self, coordinates: List[List[int]]) -> bool:
        
        n = len(coordinates)
        
        dy = coordinates[1][1] - coordinates[0][1]
        dx = coordinates[1][0] - coordinates[0][0]
        
        if dx == 0:
            return False
        
        slope = dy/dx
        
        for i in range(2,n):
            
            dy = coordinates[i][1] - coordinates[i-1][1]
            
            dx = coordinates[i][0] - coordinates[i-1][0]
            
            if dx == 0:
                
                return False
            
            if slope != (dy/dx):
                
                return False
            
            slope = dy/dx
            
        return True

File: Check_If_It_Is_A_Straight_Line.java
class Solution {
    public boolean checkStraightLine(int[][] coordinates) {
        if(coordinates.length == 2)
            return true;
        float x = coordinates[1][0] - coordinates[0][0];
        float y = coordinates[1][1] - coordinates[0][1];
        float slope = y/x;
        float b = coordinates[0][1] - (slope * coordinates[0][0]);
        for(int i = 2; i < coordinates.length; i++){
            
            float Y = coordinates[i][1];
            
            float X = coordinates[i][0];
            
            float MX = slope * X;
            
            if(Y != (MX + b))
                return false;
        }
        
        return true;
    }
}
